engine/time/src/main/java/io/deephaven/time/calendar/CalendarDay.java

```diff
@@ -30,7 +30,9 @@ public class CalendarDay<T extends Comparable<T> & Temporal> {
      */
     public static final CalendarDay<LocalTime> HOLIDAY = new CalendarDay<>();
 
-    private final List<TimeRange<T>> businessTimeRanges;
+    private final TimeRange<T>[] businessTimeRanges;
+    private final List<TimeRange<T>> businessTimeRangesList;
```

Comment:
Why have an array as well?

Reply:
To avoid breaking the API, we need to preserve `public List<TimeRange<T>> businessTimeRanges()`.   We could move the `List.of(...)` call to that method, but `List.of(...)` does work when creating the list.  
```Java
    static <E> List<E> listFromArray(E... input) {
        // copy and check manually to avoid TOCTOU
        @SuppressWarnings("unchecked")
        E[] tmp = (E[])new Object[input.length]; // implicit nullcheck of input
        for (int i = 0; i < input.length; i++) {
            tmp[i] = Objects.requireNonNull(input[i]);
        }
        return new ListN<>(tmp, false);
    }
```
It is a performance tune to avoid calling this multiple times.  It is probably not a critical perf tune for most users, so I don't have a strong view either way.